native/effektio/src/events/emoji_verification.rs

```diff
@@ -0,0 +1,577 @@
+use anyhow::Result;
+use futures::channel::mpsc::Sender;
+use log::{info, warn};
+use matrix_sdk::{
+    config::SyncSettings,
+    encryption::verification::{SasVerification, Verification},
+    ruma::{
+        events::{
+            key::verification::VerificationMethod, room::message::MessageType,
+            AnySyncMessageLikeEvent, AnyToDeviceEvent, SyncMessageLikeEvent,
+        },
+        OwnedRoomId, UserId,
+    },
+    Client,
+};
+
+use super::RUNTIME;
+
+#[derive(Clone, Debug)]
+pub struct EmojiVerificationEvent {
+    client: Client,
+    event_name: String,
+    txn_id: String,
+    sender: String,
+}
+
```

Comment:
some docs on how to use that would be nice, as this seems to be a thing that quickly becomes hard to follow again... Mind adding some inline docs here, describing a flow similar to the integration tests? Or link to some existing docs (matrix sdk example or whatever describes this nicely).

Reply:
Do you want something like this?
```rust
    /// Get information of all our own devices.
    ///
    /// # Examples
    ///
    /// ```no_run
    /// # use matrix_sdk::{Client, config::SyncSettings};
    /// # use futures::executor::block_on;
    /// # use url::Url;
    /// # use std::convert::TryFrom;
    /// # block_on(async {
    /// # let homeserver = Url::parse("http://localhost:8080")?;
    /// # let mut client = Client::new(homeserver).await?;
    /// let response = client.devices().await?;
    ///
    /// for device in response.devices {
    ///     println!(
    ///         "Device: {} {}",
    ///         device.device_id,
    ///         device.display_name.as_deref().unwrap_or("")
    ///     );
    /// }
    /// # anyhow::Ok(()) });
    /// ```
    pub async fn devices(&self) -> HttpResult<get_devices::v3::Response> {
        let request = get_devices::v3::Request::new();

        self.send(request, None).await
    }
```

Reply:
yes, but you say the SDK already has those? Then refer to that in a doc-comment. Just having that without any docs feels bad ...

Reply:
Is that okay to clone code snippet from integration test here?

Reply:
the integration tests aren't complaining about the emoji-verfication but about permission problems (addressed in #174 ). so just merge, ignoring that test for now. all non-required tests aren't in a state to be enforced yet.